src/bosh_azure_cpi/lib/cloud/azure/cloud.rb

```diff
@@ -310,21 +311,30 @@ def create_disk(size, cloud_properties, vm_cid = nil)
           disk_id = nil
           if @use_managed_disks
             if vm_cid.nil?
+              @logger.info('vm_cid is nil')
               # If instance_id is nil, the managed disk will be created in the resource group location.
               resource_group_name = _azure_config.resource_group_name
               resource_group = @azure_client.get_resource_group(resource_group_name)
               location = resource_group[:location]
               default_storage_account_type = STORAGE_ACCOUNT_TYPE_STANDARD_LRS
               zone = nil
             else
-              instance_id = InstanceId.parse(vm_cid, _azure_config.resource_group_name)
-              cloud_error('Cannot create a managed disk for a VM with unmanaged disks') unless instance_id.use_managed_disks?
-              resource_group_name = instance_id.resource_group_name
+              @logger.info("vm_cid is not nil, #{vm_cid}")
+              instance_id_obj = CloudIdParser.parse(vm_cid, _azure_config.resource_group_name)
+              cloud_error('Cannot create a managed disk for a VM with unmanaged disks') unless instance_id_obj.use_managed_disks?
+              resource_group_name = instance_id_obj.resource_group_name
               # If the instance is a managed VM, the managed disk will be created in the location of the VM.
-              vm = @azure_client.get_virtual_machine_by_name(resource_group_name, instance_id.vm_name)
-              location = vm[:location]
-              instance_type = vm[:vm_size]
-              zone = vm[:zone]
+              if instance_id_obj.is_a?(VMInstanceId)
+                vm = @azure_client.get_virtual_machine_by_name(resource_group_name, instance_id_obj.vm_name)
+                location = vm[:location]
+                instance_type = vm[:vm_size]
+                zone = vm[:zone]
+              else
+                vmss_instance = @azure_client.get_vmss_instance(resource_group_name, instance_id_obj.vmss_name, instance_id_obj.vmss_instance_id)
+                location = vmss_instance[:location]
+                instance_type = vmss_instance[:sku][:name]
+                zone = vmss_instance[:zones][0]
```

Comment:
What's the value of `zone`?
disk_manager2 will create the disks in this zone. Does it mean that VMs in VMSS are in 3 zones, but the disks are in one zone?

Reply:
the strange thing is the vm instance api.
the vm instance rest api will return an array, but there's only one value in it. 
and as we know one vm instance can only be one zone at one specific time.